spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactoryAutoConfiguration.java

```diff
@@ -52,4 +70,33 @@ public ArmeriaReactiveWebServerFactory armeriaReactiveWebServerFactory(
             ConfigurableListableBeanFactory beanFactory, Environment environment) {
         return new ArmeriaReactiveWebServerFactory(beanFactory, environment);
     }
+
+    /**
+     * Creates internal services that should not be exposed to the external network such as {@link DocService},
+     * {@link PrometheusExpositionService} and {@link HealthCheckService}.
+     *
+     * <p>Note that if a service path is either {@code null} or empty, the associated service will not be
+     * initiated. For example, {@link ArmeriaSettings#getHealthCheckPath()} is {@code null},
+     * {@link HealthCheckService} will not be created automatically.
+     *
+     * @see ArmeriaSettings#getDocsPath()
+     * @see ArmeriaSettings#getMetricsPath()
+     * @see ArmeriaSettings#getHealthCheckPath()
+     */
+    @Bean
+    public InternalServices internalServices(
```

Comment:
Question about here. Before changes, ArmeriaReactiveWebServerFactory checks if ArmeriaSettings exists and then creating InternalServices if exists. But `ArmeriaReactiveWebServerFactoryAutoConfiguration` has `EnableConfigurationProperties`, so it should be exists always? Or it's better to check if ArmeriaSettings exists here also?

```
@Bean
public Optional<InternalServices> internalServices(Optional<ArmeriaSettings>...)
```

Reply:
> Not sure if we have the case that ArmeriaSettings is not exists, like user try to disable autoconfiguration and using it manually?

It would be possible if users only use Armeria as a transport layer engine.
But I speculate that it may happen in rare cases.

Reply:
I see, my current change force user preparing InternalServices by self if they provide ArmeriaSettings without `https://armeriareactivewebserverfactoryautoconfiguration.java`. Not sure if it's ok, or I just create empty InternalServices inside(no specific management/internal setting)?

Reply:
> my current change force user preparing InternalServices by self if they provide ArmeriaSettings without https://armeriareactivewebserverfactoryautoconfiguration.java.

If I understand correctly, do you want to consider a case where users provide `ArmeriaSettings` programmatically as a `Bean` without using `application.yaml`?

Reply:
> If I understand correctly, do you want to consider a case where users provide ArmeriaSettings programmatically as a Bean without using application.yaml?

similar, so user may disable `ArmeriaReactiveWebserverFactoryAutoconfiguration.java` but providing ArmeriaSettings Bean manually. Then current code will throw NPE due to lack of InternalServcies which is used by `findBean` inside ArmeriaReactiveWebserverFactory.

Reply:
Does it make sense if we wrap `ArmeriaSettings` with `Optional`? If it is not present, we may provide port numbers through `InternalServices`?

Reply:
Ah sorry for misleading. I think this line is ok due to this configuration file force using ArmeriaSettings.java.
My concern is the change at https://github.com/line/armeria/blob/beb269691aae8ce6c23976aaccabde51c468e08e/spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java#L159-L160 need to force user preparing InternalServices Bean. Maybe ok?

Reply:
It seems OK. If users want a highly customized one, they might have some knowledge about the internal architecture. For those advanced users, we may guide them to use create it manually.

Reply:
It seems OK too. One thing that I'm concerned about is that we have duplicate code for this but I think it's OK for now. 😄

Reply:
or we may do similar thing at gradle that copy the files?

Reply:
We can do that if you don't mind. 😄

Reply:
We can take care of this, so don't worry about it. 😄

Reply:
ah forget this comment, ok if you will do it 🙇 , if not I can do it at weekend ^^

Reply:
Oh, please don't worry. Let me handle this. 😄